views/js/runner/plugins/controls/timer/timer.js

```diff
@@ -470,13 +470,51 @@ define([
                         .on('enableitem', doEnable)
                         .on('disableitem', doDisable)
                         .after('renderitem', doEnable)
+                        .before('skip', function(e, type, scope, position) {
+                            var context = testRunner.getTestContext();
+                            var testDataBeforeSkip = testRunner.getTestData();
+                            var config = testDataBeforeSkip && testDataBeforeSkip.config;
+                            var timerConfig = config && config.timer || {};
+                            var options = context && context.options || {};
+
+                            var skipPromise = new Promise(function(resolve, reject) {
+                               if (leaveTimedSection('next', scope, position) && !options.noExitTimedSectionWarning && !timerConfig.keepUpToTimeout) {
+                                    testRunner.trigger(
```

Comment:
Please get rid of duplicate code (see https://github.com/oat-sa/extension-tao-testqti/pull/1073/files#diff-998efc74cdd9f9a21d59e66f31dc4ebaR524)

Reply:
@tikhanovichA The eventifier let's you listen multiple events : 

```
.before('move skip', function(e, type, scope, position) {
```